app/src/components/modal/modal_lock_yo_tokens/index.tsx

```diff
@@ -0,0 +1,425 @@
+  color: ${props => props.theme.textfield.placeholderColor};
+`
+const DarkDataItem = styled.div`
+  display: flex;
+  align-items: center;
+  color: ${props => props.theme.colors.textColorDark};
+  font-weight: ${props => props.theme.textfield.fontWeight};
+`
+const ButtonSection = styled.div`
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  column-gap: 20px;
+  margin-top: 32px;
+`
+const ButtonsLockUnlock = styled(Button)`
+  flex: 1;
+`
+const ApproveButton = styled(ButtonStateful)`
+  flex: 1;
+`
+const ConditionalWrapper = styled.div<{ hideWrapper: boolean }>`
+  ${props => props.hideWrapper && 'display:contents'};
+  padding: 20px;
+
+  border: ${props => props.theme.borders.borderLineDisabled};
+  border-radius: ${props => props.theme.borders.commonBorderRadius};
+  margin-bottom: 24px;
+`
+const Divider = styled.div`
+  border-top: ${props => props.theme.borders.borderLineDisabled};
+  margin: 32px 0;
+`
+const PercentageText = styled.span<{ lightColor?: boolean }>`
+  ${props => props.lightColor && `color:${props.theme.colors.textColorLighter}`};
+`
+
+const ModalLockTokens = (props: Props) => {
+  const {
+    context,
+    fetchBalances,
+
+    isOpen,
+    omenBalance,
+    onClose,
+    setIsModalLockTokensOpen,
+    theme,
+  } = props
+  const { account, cpk, library: provider, networkId } = context
+
+  const omen = new OmenGuildService(provider, networkId)
+
+  const [isLockAmountOpen, setIsLockAmountOpen] = useState<boolean>(false)
+  const [displayLockAmount, setDisplayLockAmount] = useState<BigNumber>(Zero)
+  const [amountToDisplay, setAmountToDisplay] = useState<string>('')
+  const [totalLocked, setTotalLocked] = useState<BigNumber>(Zero)
+  const [userLocked, setUserLocked] = useState<BigNumber>(Zero)
+  const [timestamp, setTimestamp] = useState<number>(0)
+  const [unlockTimeLeft, setUnlockTimeLeft] = useState<number>(0)
+  const [isTransactionModalOpen, setIsTransactionModalOpen] = useState<boolean>(false)
+  const [transactionMessage, setTransactionMessage] = useState<string>('')
+  const [txHash, setTxHash] = useState<any>('')
+  const [txState, setTxState] = useState<TransactionStep>(TransactionStep.idle)
+  const [txNetId, setTxNetId] = useState()
+  const [omenAllowance, setOmenAllowance] = useState<BigNumber>(Zero)
+  const [allowanceState, setAllowanceState] = useState<ButtonStates>(ButtonStates.idle)
+
+  const isApproveVisible =
+    (omenAllowance.isZero() && isLockAmountOpen) ||
+    (allowanceState === ButtonStates.finished && !omenAllowance.isZero())
+
+  useEffect(() => {
+    setAllowanceState(ButtonStates.idle)
+    ;(async () => {
+      await getTokenLockInfo()
+      await fetchAllowance()
+    })()
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [networkId, account, cpk])
+
+  const getTokenLockInfo = async () => {
+    try {
+      if (cpk?.address) {
+        let address
+        if (context.networkId === networkIds.MAINNET && !context.relay) {
```

Comment:
Why is this only on mainnet?

Reply:
So that is because other then on mainnet we are checking cpk address for the votes user has and only on mainnet users address. This needs to be checked how it will work with cpk once we deploy it somewhere other then rinkeby...